### ISSUE TITLE
[macOS] Native Screen Recording Stops Winit Recording KB Input

### ISSUE THREAD

**Issue by sanbox-irl** [rust-windowing/winit#1465]
Hey there,

When you use native macOS recording (Command Shift 5) to record the screen, winit stops detecting keyboard input entirely. In fact, it basically stops until the applicaiton is completely restarted! Clicking in and out of the window doesn't seem to help.

This makes sharing gifs and videos of tools in development pretty tough. 

I have a mac (obviously) and would be more than happy to help if i could get a pointer to where to start? Winit's a bigger codebase so it can be a little overwhelming...

**Comment by sanboxrunner** [comment]
oh no this doesn't look that bad...I can totally figure this out. i'll try to fix it and see if I can add a PR when I do

**Comment by sanbox-irl** [comment]
I have investigated the issue and have discovered, I believe, the problem.

The issue was that in my code, I was checking modifier change states using normal KeyboardInput events (ie, checking if RWin or SHIFT keys had been pressed or released).

The problem is that if you do Command (RWIN on this board) + SHIFT + 3 or 4, you trigger the screencapture or video capture widget. If you release command / shift while that widget is active but BEFORE deactiving the widget (release mouse on 3, ending recording on 4), then MacOS will notify winit that it has changed those keys simply by sending two 0x00 scancodes. 

This may be why ModifiersEvent has been added -- it will correctly inform that the Modifiers have been changed, tho if anyone is reading this, beware you won't get any keyboard released events!

Since this is a fairly common operation, do we want to make a synthetic "released" event when we know that such an event has occurred? 

I'm adding a doc comment right now to the modifier event so future programmers may suffer less
